src/main/java/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/machines/ReactorAccessPort.java

```diff
@@ -39,39 +41,29 @@ public void init() {
 
 			@Override
 			public boolean canOpen(Block b, Player p) {
-				if(p.hasPermission("slimefun.inventory.bypass") || SlimefunPlugin.getProtectionManager().hasPermission(p, b.getLocation(), ProtectableAction.ACCESS_INVENTORIES)) {
-					AReactor reactor = getReactor(b.getLocation());
-					if (reactor != null) {
-						boolean empty = true;
-						BlockMenu bm = getReactorMenu(b.getLocation());
-						
-						if (bm != null) {
-							for (int slot: reactor.getCoolantSlots()) {
-								if (bm.getItemInSlot(slot) != null) {
-									empty = false;
-								}
-							}
-							
-							for (int slot: reactor.getFuelSlots()) {
-								if (bm.getItemInSlot(slot) != null) {
-									empty = false;
-								}
-							}
-
-							if(!empty || !p.isSneaking()) {
-								//reactor is not empty, lets view it's inventory instead.
-								bm.open(p);
-								return false;
-							}
-							
-						}
-						
-						return true;
-
-					} 
+				return p.hasPermission("slimefun.inventory.bypass") || SlimefunPlugin.getProtectionManager().hasPermission(p, b.getLocation(), ProtectableAction.ACCESS_INVENTORIES);
+			}
+			
+			@Override
+			public void newInstance(final BlockMenu menu, final Block b) {
+				BlockMenu reactor = getReactorMenu(b.getLocation());
+				if (reactor != null) {
+					menu.replaceExistingItem(INFO_SLOT, new CustomItem(new ItemStack(Material.GREEN_WOOL), "&7Reactor", "", "&6Detected", "", "&7> Click to view Reactor"));
+					menu.addMenuClickHandler(INFO_SLOT, (p, slot, item, action) -> {
+						reactor.open(p);
```

Comment:
This could potentially cause a NullPointerException if the Reactor was broken.
So there needs to be a null-check here!